ckanext/switzerland/helpers/format_utils.py

```diff
@@ -86,13 +95,18 @@ def get_resource_format(media_type, format, download_url):
     return 'SERVICE'
 
 
-def prepare_formats_for_index(resources):
-    """generates a set with formats of all resources"""
+def prepare_formats_for_index(resources, linked_data_only=False):
+    """Generate a deduplicated list with formats of all resources, or all
+    Linked Data formats that are present.
+    """
     formats = set()
     for r in resources:
         resource = prepare_resource_format(
             resource=r
         )
+        if linked_data_only \
```

Comment:
hmm, I stumbled over this as it felt a little weird to me, to check for `linked_data_only` this way to then ignore everything else. instead of specifying `all`, `linked_data_only`, `common_data_only` or something along those lines. but honestly, not sure if it would be worth it, to refactor it anyways, but in case you already thought about this beforehand, maybe worth discussing LMK :)

Reply:
To me this method seems cleaner - I like the "early return" pattern, where you check early if you can return out of a method/function/loop item. If we have `linked_data_only=True` and the format is not one of the linked data formats, there's nothing else we need to consider because we already know we won't add the data.

I guess the alternative would be checking this in both the `if` and the `else` below, which seems messier to me. Or am I missing something?

Reply:
sorry, I think I didn't explain it clear enough 😃 
I like the return early method!
I was about to write some dummy code to try to explain it but I think my thoughts wouldn't add much value. Will try to explain what I meant tomorrow, but feel free to merge as is 😄